Library/SharedFunctions.swift

```diff
@@ -174,3 +174,19 @@ public func ksr_isOSVersionAvailable(_ version: Double) -> Bool {
 
   return false
 }
+
+public typealias AttributedLinkData = (text: String, url: URL?, attributes: [NSAttributedString.Key: Any])
+
+public func ksr_attributedString(_ attributedString: NSAttributedString,
+                                 with links: [AttributedLinkData]) -> NSAttributedString {
+  let string = attributedString.string as NSString
+  let mutableAttributedString = NSMutableAttributedString(attributedString: attributedString)
+
+  links.map { text, url, attributes -> (attributes: [NSAttributedString.Key: Any], range: NSRange) in
+    (attributes.merging([.link: url as Any]) { $1 }, string.range(of: text))
```

Comment:
Merges the link attribute, it's respective URL and any other attributes that are passed in for that particular text.

Reply:
I would if using `localizedStandardRange(of str: String)` might be safer here?